vlab-research/fly
# Optimise abstraction of replybot into web survey

### What changes have you made?
- Refactored `form.js` and `form.test.js` to be a duplicate of the code in `replybot`
- Added in any extra unit tests that do not currently exist in `replybot` 
- Reset the `sample.json` 
- Updated `responsStore.js` and `responseStore.test.js` in line with these changes
- Removed any instances of field type `legal` as we've since decided to deprecate this feature
- Merged #93 into this branch so that it includes components of additional field types `opinion_scale` and `rating`

### Which issue(s) does this PR address?
#88 

### How to test
- Run npm install and open up the web survey on `localhost:5000`
- Check that all tests are passing
- You should be able to switch around the data by importing also the JSON from `test-1` and `test-2` without any breaks in tests or functionality

### Todo
- Create a new version of the `validator` module that is more in line with the needs of the web survey
